lint-staged.config.js

```diff
@@ -1,4 +1,4 @@
 module.exports = {
-  '**/*.ts?(x)': () => 'tsc -p tsconfig.json --noEmit',
```

Comment:
It didn't make sense to check the types of the whole project before committing, it's a slow task and we are interested in getting quick feedback at the pre-commit stage.

Reply:
That's okay, so long as we do run the full suite of tests before merging into `main`. 😉
So probably avoid pushing things to `main` directly. 😄

Reply:
Yes, `main` branch is protected and types will be checked on CI.